py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -0,0 +1,486 @@
+"""This module defines the class Dr16ExpectedFlux"""
+import logging
+import multiprocessing
+
+import fitsio
+from astropy.io import fits
+import numpy as np
+from scipy.interpolate import interp1d
+import healpy
+from pkg_resources import resource_filename
+
+from picca.delta_extraction.astronomical_objects.forest import Forest
+from picca.delta_extraction.astronomical_objects.pk1d_forest import Pk1dForest
+from picca.delta_extraction.errors import ExpectedFluxError
+from picca.delta_extraction.expected_flux import ExpectedFlux
+
+accepted_options = ["input directory", "iter out prefix",
+                    "num processors",
+                    "var lss binning"]
+
+defaults = {
+    "iter out prefix": "delta_attributes",
+}
+
+
+class TrueContinuum(ExpectedFlux):
+    """Class to the expected flux calculated used the true continuum unabsorbed contiuum
+    for mocks. It uses var_lss pre-computed from mocks and the mean flux estimated from 
+
+    Methods
+    -------
+    extract_deltas (from ExpectedFlux)
+    __init__
+    _initialize_arrays_lin
+    _initialize_arrays_log
+    _parse_config
+    compute_delta_stack
+    compute_expected_flux
+    read_true_continuum
+    read_var_lss
+    populate_los_ids
+
+    Attributes
+    ----------
+    los_ids: dict (from ExpectedFlux)
+    A dictionary to store the mean expected flux fraction, the weights, and
+    the inverse variance for each line of sight. Keys are the identifier for the
+    line of sight and values are dictionaries with the keys "mean expected flux",
+    and "weights" pointing to the respective arrays. If the given Forests are
+    also Pk1dForests, then the key "ivar" must be available. Arrays have the same
+    size as the flux array for the corresponding line of sight forest instance.
+
+
+    get_stack_delta: scipy.interpolate.interp1d or None
+    Interpolation function to compute the mean delta (from stacking all lines of
+    sight). None for no info.
+
+    get_var_lss: scipy.interpolate.interp1d
+    Interpolation function to compute mapping functions var_lss. See equation 4 of
+    du Mas des Bourboux et al. 2020 for details. Data for interpolation is read from a file.
+
+    input_directory: str
+    Directory where true continum data is store
```

Comment:
I would have thought that the directory with the true continuum is the same directory where the mock spectra are. Do we need another variable?

Reply:
Ignasi confirmed that indeed they are the same directory, that we should synchronise at some point (will open issue)

Reply:
I think it could be useful to keep it at least as optional to pass a different directory, I can imagine running some mocks and not used the true continuum flag, but having the same mock with different specifications but same true continuum so I can use it. But as you prefer.